### ISSUE TITLE
calico-kubernetes doesn't create profiles for endpoints in AWS

### ISSUE THREAD

**Issue by geriBatai** [projectcalico/calico-kubernetes#63]
Hi,

I have a similar issue to #44. Running calico-kubernetes with kubernetes-1.1.0-alpha1, tried calico-node 0.7.0 and 0.6.0, docker version 1.8.0. Manual calico-docker setups work, but running kubelet with calico network plugin does not setup any profiles (networking works if profile is created manually and is attached to the endpoint).

```
# calicoctl endpoint show --detailed
+---------------------------+-----------------+------------------------------------------------------------------+----------------------------------+-----------------+-------------------+----------+--------+
|          Hostname         | Orchestrator ID |                           Workload ID                            |           Endpoint ID            |    Addresses    |        MAC        | Profiles | State  |
+---------------------------+-----------------+------------------------------------------------------------------+----------------------------------+-----------------+-------------------+----------+--------+
| kubernetes-minion-simas-1 |      docker     | 4f0d7178638c23359c231d51141e46f35cf0916081ddc309553b2dcc5fbdd44e | c84cd2486c2711e584790e23dbe0df45 |  192.168.1.3/32 | 02:33:72:8e:04:e1 |          | active |
| kubernetes-minion-simas-1 |      docker     | f965cc1a77642cc0738fe83c2a52fe6f5c2e001edec2a29a63008068c621c1fd | c7af6d6e6c2711e59f020e23dbe0df45 |  192.168.1.2/32 | 3e:39:5c:3a:81:60 |          | active |
| kubernetes-minion-simas-2 |      docker     | 143c5e13c39299abfe8643369a7ec864855623074f390cde202ab16a32dd165f | c9ee62066c2711e5971c0e46fb8fddff |  192.168.2.2/32 | 12:ee:9e:cd:d7:7c |          | active |
| kubernetes-minion-simas-2 |      docker     | 7ab5e1ec14c365e3fb0b539218118677387d3ad17bacecfd3d36462dfe2cc7db | c9ba880a6c2711e59f0a0e46fb8fddff |  192.168.2.3/32 | 62:4a:0e:e2:33:2c |          | active |
| kubernetes-minion-simas-2 |      docker     | fe049aea7d841614a62ebec3eb6287d9cdba8c2ce4d5d0208e2b44fc89877520 | 0cf527846c2c11e5b79d0e46fb8fddff | 192.168.2.99/32 | e6:4b:6f:2c:31:0e |          | active |
+---------------------------+-----------------+------------------------------------------------------------------+----------------------------------+-----------------+-------------------+----------+--------+

# calicoctl profile show --detailed
+------+------+-----------------+-------------+-------------+-------+
| Name | Host | Orchestrator ID | Workload ID | Endpoint ID | State |
+------+------+-----------------+-------------+-------------+-------+
+------+------+-----------------+-------------+-------------+-------+
```

kubelet.log can be found at https://gist.github.com/geriBatai/1ac724b4da266d4d513d

**Comment by Symmetric** [comment]
Looks like we're somehow failing to set up networking in the Pod infrastructure container:

```
Oct 06 12:42:13 kubernetes-minion-simas-2 kubelet[466]: Error code 2 creating pod networking:
Oct 06 12:42:13 kubernetes-minion-simas-2 kubelet[466]: Command '['ip', 'netns', 'exec', '2190', 'ip', 'link', 'del', 'eth0']' returned non-zero exit status 2
```

Unfortunately the plugin logs in that area aren't very good, so I've cut a new release which beefs up the logging.

Can you try again with [version 0.2.1](https://github.com/projectcalico/calico-kubernetes/releases/tag/v0.2.1) of the calic-kubernetes plugin?

Make sure you have the `LOG_LEVEL=debug` env variable set in the kubelet's environment (defaults to INFO, and at that level you won't get the logs I just added), and please upload a gist of i) the kubelet.log file, ii) the `/var/log/calico/kubernetes/calico.log` file.

**Comment by biwwy** [comment]
I'm not sure if I'm having the same issue, but I have similar symptoms. No profiles are created authomatically.

Below is the logs:

https://gist.github.com/biwwy/f87d077c05a7bb343f2a

Tried with calicoctl 0.8.0 and calico-kubernetes 0.2.1. 
CoreOS both stable and alpha channels. 
Docker 1.7.6 and 1.8.3. 
Kubelet v1.1.0-alpha.1.1596+99e260530226b4, but all other Kubernetes components (Proxy, API-server) are 1.0.6 from hyperkube image.

**Comment by caseydavenport** [comment]
@biwwy - It looks like you're hitting a bug fixed in PR #74.  We've just released version v0.3.0 of the plugin, which includes this fix.  You can download it here: https://github.com/projectcalico/calico-kubernetes/releases/tag/untagged-b46344ad94f2c81584cb

Let me know if this fixes your problem!

**Comment by caseydavenport** [comment]
@biwwy - I just noticed that release v0.3.0 is still a draft, so you likely can't access it yet.  I suspect it will be officially released within the next day.  I'll report back here when it is!

**Comment by caseydavenport** [comment]
@biwwy - v0.3.0 has been released.

**Comment by biwwy** [comment]
@caseydavenport Thanks! The error about _log_interfaces function is gone, still I don't see profile created, though looks like for different reasons. More logs below:

https://gist.github.com/biwwy/0e33d2542bc06a91c55e

**Comment by caseydavenport** [comment]
@biwwy - Thanks for those logs.  I'll take a look at this today.

It appears as if the containers netns disappears partway through execution, as shown by these logs:
```
2015-10-23 11:41:26,905 7362 DEBUG Namespaces:
7351
1557
1551
1294
1287
1288
1282
```
and then later
```
2015-10-23 11:41:27,568 7362 DEBUG Namespaces:
1557
1551
1294
1287
1288
1282
```

I'll try to reproduce this and report back.

**Comment by caseydavenport** [comment]
@biwwy - I managed to reproduce what you're seeing - it looks like we're seeing this issue: http://stackoverflow.com/questions/31265993/docker-networking-namespace-not-visible-in-ip-netns-list

The error you're hitting is only in logging, and should only occur when DEBUG logging is enabled in the plugin.

Setting `LOG_LEVEL=INFO` should work around the problem for now - let me know if this works for you.

I've raised issue #81 to track this and will get a fix in for the next release.
